Assets/Tests/Weaver/Assembler.cs

```diff
@@ -45,139 +44,82 @@ public InMemoryAssembly InMemoryAssembly
+        readonly HashSet<string> sourceFiles = new HashSet<string>();
+
+        public Assembler()
         {
-            SourceFiles = new HashSet<string>();
-            ReferenceAssemblies = new HashSet<string>();
             CompilerMessages = new List<CompilerMessage>();
         }
 
         // Add a range of source files to compile
-        public static void AddSourceFiles(string[] sourceFiles)
+        public void AddSourceFiles(string[] sourceFiles)
         {
             foreach (string src in sourceFiles)
             {
-                SourceFiles.Add(Path.Combine(OutputDirectory, src));
-            }
-        }
-
-        // Find reference assembly specified by asmName and store its full path in asmFullPath
-        // do not pass in paths in asmName, just assembly names
-        public static bool FindReferenceAssemblyPath(string asmName, out string asmFullPath)
-        {
-            asmFullPath = "";
-
-            Assembly[] asms = CompilationPipeline.GetAssemblies();
-            foreach (Assembly asm in asms)
-            {
-                foreach (string asmRef in asm.compiledAssemblyReferences)
-                {
-                    if (asmRef.EndsWith(asmName))
-                    {
-                        asmFullPath = asmRef;
-                        return true;
-                    }
-                }
+                this.sourceFiles.Add(Path.Combine(WeaverTestLocator.OutputDirectory, src));
             }
-
-            return false;
-        }
-
-        // Add reference (not cleared during calls to Clear)
-        public static void ClearReferences()
-        {
-            ReferenceAssemblies.Clear();
         }
 
         // Delete output dll / pdb / mdb
-        public static void DeleteOutput()
+        public void DeleteOutput()
         {
             // "x.dll" shortest possible dll name
-            if (OutputFile.Length < 5)
+            if (WeaverTestLocator.OutputFile.Length < 5)
             {
                 return;
             }
 
-            string projPathFile = Path.Combine(OutputDirectory, OutputFile);
-
             try
             {
-                File.Delete(projPathFile);
+                File.Delete(ProjectPathFile);
             }
             catch { /* Do Nothing */ }
 
             try
             {
-                File.Delete(Path.ChangeExtension(projPathFile, ".pdb"));
+                File.Delete(Path.ChangeExtension(ProjectPathFile, ".pdb"));
             }
             catch { /* Do Nothing */ }
 
             try
             {
-                File.Delete(Path.ChangeExtension(projPathFile, ".dll.mdb"));
+                File.Delete(Path.ChangeExtension(ProjectPathFile, ".dll.mdb"));
             }
             catch { /* Do Nothing */ }
         }
 
         // clear all settings except for referenced assemblies (which are cleared with ClearReferences)
-        public static void Clear()
+        public void Clear(bool deleteOutputOnClear = true)
```

Comment:
I dont think we need to clean up state any more, as we can just create a new instance instead. I think the test class can instead call `DeleteOutput`

Reply:
Assembler is created only once, not every time [SetUp] is called. Can change that if you prefer

Reply:
I think it would be better to create a new instance in side setup, so we never have to worry about cleaning up state